src/codeu/chat/server/Controller.java

```diff
@@ -34,9 +35,38 @@
   private final Model model;
   private final Uuid.Generator uuidGenerator;
 
+  private final LocalFile localFile;
+
+  private boolean isInitialized = false;
+
   public Controller(Uuid serverId, Model model) {
     this.model = model;
     this.uuidGenerator = new RandomUuidGenerator(serverId, System.currentTimeMillis());
+    this.localFile = null;
+  }
+  //New constructor, which can get the local file information.
+  public Controller(Uuid serverId, Model model,LocalFile localFile) {
+    this.model = model;
+    this.uuidGenerator = new RandomUuidGenerator(serverId, System.currentTimeMillis());
+    
+    this.localFile = localFile;//The path is assigned by server.
+    //Load the data from local file
+    for (User item : localFile.getCopyOfUsers())
+    {
+      this.newUser(item.id, item.name, item.creation);
+    }
+
+    for (ConversationHeader item : localFile.getCopyOfConversationHeaders())
+    {
+      this.newConversation(item.id, item.title, item.owner, item.creation);
+    }
+
+    for(Message item :localFile.getCopyOfMessages())
+    {
+      this.newMessage(item.id, item.author, item.conversation, item.content, item.creation);
+    }
+    //Initialization Complete
+    isInitialized = true;
```

Comment:
Why do you have this variable?  It looks like it will always be true, no matter what, which means you should get rid of it.